api/v2/api.go

```diff
@@ -672,3 +672,31 @@ func parseFilter(filter []string) ([]*labels.Matcher, error) {
 	}
 	return matchers, nil
 }
+
+var (
+	swaggerSpecCacheMx sync.Mutex
```

Comment:
We should be able to make this a `RWMutex`, although I'm not sure if we would get much out of it?

Reply:
I don't think the extra complexity of handling a RWMutex (first read lock to check if it's cached, then take write lock to cache it) is worth. We don't expect this lock to be hit very frequently (just once in Prometheus Alertmanager).